custom_components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,545 @@
+"""Support for Z-Wave climate devices."""
+from typing import Optional, Tuple
+
+from openzwavemqtt.const import CommandClass
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_FAN,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_BOOST,
+    PRESET_NONE,
+    SUPPORT_AUX_HEAT,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_TARGET_TEMPERATURE_RANGE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DOMAIN
+from .entity import ZWaveDeviceEntity
+
+REMOTEC = 0x5254
+REMOTEC_ZXT_120 = 0x8377
+REMOTEC_ZXT_120_THERMOSTAT = (REMOTEC, REMOTEC_ZXT_120)
+ATTR_OPERATING_STATE = "operating_state"
+ATTR_FAN_STATE = "fan_state"
+ATTR_FAN_ACTION = "fan_action"
+AUX_HEAT_ZWAVE_MODE = "Aux Heat"
+
+# Device is in manufacturer specific mode (e.g. setting the valve manually)
+PRESET_MANUFACTURER_SPECIFIC = "Manufacturer Specific"
+
+WORKAROUND_ZXT_120 = "zxt_120"
+
+DEVICE_MAPPINGS = {REMOTEC_ZXT_120_THERMOSTAT: WORKAROUND_ZXT_120}
+
+HVAC_STATE_MAPPINGS = {
+    "off": HVAC_MODE_OFF,
+    "heat": HVAC_MODE_HEAT,
+    "heat mode": HVAC_MODE_HEAT,
+    "heat (default)": HVAC_MODE_HEAT,
+    "furnace": HVAC_MODE_HEAT,
+    "fan only": HVAC_MODE_FAN_ONLY,
+    "dry air": HVAC_MODE_DRY,
+    "moist air": HVAC_MODE_DRY,
+    "cool": HVAC_MODE_COOL,
+    "heat_cool": HVAC_MODE_HEAT_COOL,
+    "auto": HVAC_MODE_HEAT_COOL,
+    "auto changeover": HVAC_MODE_HEAT_COOL,
+}
+
+MODE_SETPOINT_MAPPINGS = {
```

Comment:
You should replace the strings with the integer values. This means it's unnecessary to account for any custom values in the XML files.